packages/db/src/search/ExecutionSearch.ts

```diff
@@ -0,0 +1,129 @@
+    } = TableNames;
+
+    const searchQuery = knex(`${this.tableName} as ${this.tableName}`)
+      .leftJoin(`${this.tableName} as ${this.tableName}-parent`, `${this.tableName}.parent_cumulus_id`, `${this.tableName}-parent.cumulus_id`)
+      .select(`${this.tableName}.*`)
+      .select({
+        collectionName: `${collectionsTable}.name`,
+        collectionVersion: `${collectionsTable}.version`,
+        asyncOperationId: `${asyncOperationsTable}.id`,
+        parentArn: `${this.tableName}-parent.arn`,
+      });
+
+    const countQuery = knex(this.tableName)
+      .count(`${this.tableName}.cumulus_id`);
+
+    if (this.searchCollection()) {
+      countQuery.innerJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+      searchQuery.innerJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+    }
+
+    if (this.searchAsync()) {
+      countQuery.innerJoin(asyncOperationsTable, `${this.tableName}.async_operation_cumulus_id`, `${asyncOperationsTable}.cumulus_id`);
+      searchQuery.innerJoin(asyncOperationsTable, `${this.tableName}.async_operation_cumulus_id`, `${asyncOperationsTable}.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(asyncOperationsTable, `${this.tableName}.async_operation_cumulus_id`, `${asyncOperationsTable}.cumulus_id`);
+    }
+    return { countQuery, searchQuery };
+  }
+
+  /**
+   * Build queries for infix and prefix
+   *
+   * @param params
+   * @param params.countQuery - query builder for getting count
+   * @param params.searchQuery - query builder for search
+   * @param [params.dbQueryParameters] - db query parameters
+   */
+  protected buildInfixPrefixQuery(params: {
+    countQuery: Knex.QueryBuilder,
+    searchQuery: Knex.QueryBuilder,
+    dbQueryParameters?: DbQueryParameters,
+  }) {
+    const { countQuery, searchQuery, dbQueryParameters } = params;
+    const { infix, prefix } = dbQueryParameters ?? this.dbQueryParameters;
+    if (infix) {
+      countQuery.whereLike(`${this.tableName}.arn`, `%${infix}%`);
+      searchQuery.whereLike(`${this.tableName}.arn`, `%${infix}%`);
+    }
+    if (prefix) {
+      countQuery.whereLike(`${this.tableName}.arn`, `${prefix}%`);
+      searchQuery.whereLike(`${this.tableName}.arn`, `${prefix}%`);
+    }
+  }
+
+  /**
+   * Translate postgres records to api records
+   *
+   * @param pgRecords - postgres records returned from query
+   * @returns translated api records
+   */
+  protected translatePostgresRecordsToApiRecords(pgRecords: ExecutionRecord[])
+    : Partial<ApiExecutionRecord[]> {
+    log.debug(`translatePostgresRecordsToApiRecords number of records ${pgRecords.length} `);
+    const results: ApiExecutionRecord[] = [];
+    pgRecords.map((executionRecord) => {
+      const collectionPgRecord = {
```

Comment:
You can check if collection_cumulus_id is set, if not collectionId is undefined.